agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -125,6 +134,61 @@ func (state *DockerTaskEngineState) allImageStates() []*image.ImageState {
 	return allImageStates
 }
 
+func (state *DockerTaskEngineState) AllENIAttachments() []*api.ENIAttachment {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.allENIAttachments()
+}
+
+func (state *DockerTaskEngineState) allENIAttachments() []*api.ENIAttachment {
+	var allENIAttachments []*api.ENIAttachment
+	for _, v := range state.eniAttachments {
+		allENIAttachments = append(allENIAttachments, v)
+	}
+
+	return allENIAttachments
+}
+
+func (state *DockerTaskEngineState) ENIByMac(mac string) (*api.ENIAttachment, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	eni, ok := state.eniAttachments[mac]
+	return eni, ok
+}
+
+func (state *DockerTaskEngineState) AddENIAttachment(eniAttachment *api.ENIAttachment) {
+	if eniAttachment == nil {
+		log.Debug("Cannot add empty eni attachment information")
+		return
+	}
+
+	state.lock.Lock()
+	defer state.lock.Unlock()
+
+	if _, ok := state.eniAttachments[eniAttachment.MacAddress]; !ok {
+		state.eniAttachments[eniAttachment.MacAddress] = eniAttachment
+	} else {
+		seelog.Debugf("Duplicate eni attachment information: %v", eniAttachment)
+	}
+
+}
+func (state *DockerTaskEngineState) RemoveENIAttachment(mac string) {
+	if mac == "" {
+		log.Debug("Cannot remove empty eni attachment information")
+		return
+	}
+	state.lock.Lock()
+	defer state.lock.Unlock()
+
+	if _, ok := state.eniAttachments[mac]; ok {
```

Comment:
I don't think you need to do this. Just invoking `delete` should be fine. Go handles deletion of non existing keys by essentially not doing anything.